_layouts/docwithnav.html

```diff
@@ -186,8 +149,39 @@
 			}
 		    return false; // don't actually follow the link (avoid reloading page (default browser click behavior))
 	    });
+		/* Load the contents of the target page. Load the dynamic TOC otherwise scroll if there's an anchor */
+		function loadandscrollpage(targethref){
+			var navitem = findcurnavitem(),
+				offset = 0; //set default offset to the top of the page
+			$('#doccontent').load(targethref + ' #doccontent > *', function() {
+				if (navitem.hasClass('gentoc')) { //load the in-page TOC from the nav menu
+					var tocid = String(navitem.parent().children('ul').attr('id'));//cast id to string
+					$('#toclist').load( location.pathname + " #" + tocid + " li" );
+				} else if (targethref.indexOf('#') > 0) {//change offset if URL has anchor
+					var hash = "#" + targethref.substring(targethref.indexOf("#")+1);//retrieve only the anchor from the URL
+					if ($(hash).length) {//make sure its not an empty anchor
+						offset = $(hash).offset().top - 155;
+					}
+				}
+				$('html, body').animate({scrollTop: offset}, 0);
+			});
+		}
+		/* Load the code samples in a syntax formatted code block */
```

Comment:
i also consolidated this behavior into a single function (changed nothing but moved two similar instances of this code into one function)